lean/common.lean

```diff
@@ -297,6 +307,18 @@ def rax := reg64 0
 def rcx := reg64 1
 def rdx := reg64 2
 def rbx := reg64 3
+def rsi := reg64 4
+def rdi := reg64 5
+def rsp := reg64 6
+def rbp := reg64 7
```

Comment:
I prefer the numbering scheme used in encoding registers in machine code, so it should be:

```
def rsp := reg64 4
def rbp := reg64 5
def rsi := reg64 6
def rdi := reg64 7
```

See http://www.c-jump.com/CIS77/CPU/x86/X77_0060_mod_reg_r_m_byte.htm